layers/exporters/svg_objects.py

```diff
@@ -0,0 +1,320 @@
+import drawSvg as draw
+import string
+import colorsys
+
+def _getapproximatestringwidth(st):
+    """
+        Internal - estimate the width of a string... somewhat poorly
+
+        :param st: The string to evaluate
+        :return: A number that roughly describes the length of the string
+    """
+    size = 0
+    for s in st:
+        if s in 'lij|\' ':
+            size += 37
+        elif s in '![]fI.,:;/\\t':
+            size += 50
+        elif s in '`-(){}r"':
+            size += 60
+        elif s in '*^zcsJkvxy':
+            size += 85
+        elif s in 'aebdhnopqug#$L+<>=?_~FZT' + string.digits:
+            size += 95
+        elif s in 'BSPEAKVXY&UwNRCHD':
+            size += 112
+        elif s in 'QGOMm%W@':
+            size += 135
+        else:
+            size += 50
+    return size * 6 / 75
+
```

Comment:
Maybe this would be a more accurate option? https://stackoverflow.com/a/35772222

We'll probably need to include the `ttf` files for our typefaces in the directory to make this work, I wasn't able to instantiate an ImageFont instance otherwise. 

This also may allow implementation of [this function](https://github.com/mitre-attack/attack-navigator/blob/master/nav-app/src/app/exporter/exporter.component.ts#L248-L292) from the Navigator, which would improve the UX of this export by optimizing font size automatically (currently some techniques have spillover). That function basically just tries every combination of wrapping where spaces occur in the text to determine which line break locations result in the optimum possible font size without overflowing the container bounds.

Reply:
Although now that I'm looking at the implementation below, it looks like you already implemented a version of that function. Perhaps improving the measurement of text size using the linked function would still improve behavior (my tests saw a few areas of text exceeding the container):

![image](https://user-images.githubusercontent.com/28868888/89535859-03c51c00-d7c5-11ea-9558-14c1a14d523c.png)

Reply:
I'll look into pillow as an potion for the spacing. I ended up manually creating and tweaking the measurement and wrapping functions to decent effect, but as you found, it's not perfect.